### PR TITLE
Do not modify existing input alarm messages if mbox is full

### PR DESCRIPTION
**Pull request by pyhys**

### CODE REVIEW

File: src/common/pf_alarm.c [hunk 1]
```diff
@@ -704,53 +704,64 @@ static int pf_alarm_apmr_frame_handler (
...
-      if (p_apmx->apmr_msg_nbr >= NELEMENTS (p_apmx->apmr_msg))
-      {
-         p_apmx->apmr_msg_nbr = 0;
-      }
-      p_apmr_msg = &p_apmx->apmr_msg[nbr];
-      p_apmr_msg->p_buf = p_buf;
-      p_apmr_msg->frame_id_pos = frame_id_pos;
       if (p_apmx->p_alarm_q != NULL)
       {
+         nbr = p_apmx->apmr_msg_nbr++; /* ToDo: Make atomic */
```
Comment (olbjo) [inline src/common/pf_alarm.c:714]: Remove ToDo?
Can p_apmx->apmr_msg_nbr be updated when post succeeds instead?

File: src/common/pf_alarm.c [hunk 2]
```diff
@@ -704,53 +704,64 @@ static int pf_alarm_apmr_frame_handler (
...
+               __LINE__,
+               p_apmx->high_priority ? "high" : "low",
+               p_buf);
+
             ret = 1; /* Means that calling function should not free buffer,
                         as that will be done when reading the mbox */
          }
          else
          {
+            /* Undo modification to apmx */
```
Comment (olbjo) [inline src/common/pf_alarm.c:738]: Maybe there is a way to instead "commit" modification to apmx when the "post" operation succeeds instead?

**Review by olbjo — APPROVED** [review]

File: src/common/pf_alarm.c (hunk 1 above)
Comment (pyhys) [reply src/common/pf_alarm.c]: Now updated when post succeeds
Kept TODO
Done

File: src/common/pf_alarm.c [hunk 3]
```diff
@@ -704,53 +704,64 @@ static int pf_alarm_apmr_frame_handler (
...
+         nbr = p_apmx->apmr_msg_nbr++; /* ToDo: Make atomic */
+         if (p_apmx->apmr_msg_nbr >= NELEMENTS (p_apmx->apmr_msg))
+         {
+            p_apmx->apmr_msg_nbr = 0;
+         }
+         p_apmr_msg = &p_apmx->apmr_msg[nbr];
+
          if (os_mbox_post (p_apmx->p_alarm_q, (void *)p_apmr_msg, 0) == 0)
          {
+            p_apmr_msg->p_buf = p_buf;
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/common/pf_alarm.c:719]: Added a TODO about the possible race condition

File: src/common/pf_alarm.c (hunk 2 above)
Comment (pyhys) [reply src/common/pf_alarm.c]: Now updated `apmr_msg_nbr` when post succeeds
Done

**Pull request merged**